addons/account/models/account.py

```diff
@@ -1958,6 +1958,14 @@ class AccountTaxRepartitionLine(models.Model):
     country_id = fields.Many2one(string="Country", comodel_name='res.country', related='company_id.country_id',  help="Technical field used to restrict tags domain in form view.")
     company_id = fields.Many2one(string="Company", comodel_name='res.company', required=True, default=lambda x: x.env.company, help="The company this repartition line belongs to.")
     sequence = fields.Integer(string="Sequence", default=1, help="The order in which display and match repartition lines. For refunds to work properly, invoice repartition lines should be arranged in the same order as the credit note repartition lines they correspond to.")
+    use_in_tax_closing = fields.Boolean(string="Tax Closing Entry")
+
+    @api.onchange('account_id')
+    def _on_change_account_id(self):
+        if not self.account_id:
+            self.use_in_tax_closing = False
+        else:
+            self.use_in_tax_closing = not(self.account_id.internal_group == 'income' or self.account_id.internal_group == 'expense')
```

Comment:
@oco-odoo @bfr-o 
Just questioning myself here. Will it work if we call create with the following values?
{
'account_id': 'anything in income',
'use_in_tax_closing': True
}
Will the value stays True or will it be reset by the onchange to False? If it's the latter, we've got a problem

Also we might want to add a migration script to add default value to that new field what do you think?

Reply:
Yes it should remain as True until you change the account_id, then the default (computed) value will override the user set value, but that makes sense in most cases.

I've updated the migration script.